client/app/domain/zone/record/domain-zone-record.controller.js

```diff
@@ -442,6 +450,15 @@ angular.module("App").controller(
             }
         }
 
+        loadCaaModel (target) {
+            const splitted = target.match(this.DomainValidator.regex.CAA);
+            if (_.isArray(splitted) && splitted.length > 0) {
+                this.model.target.flags = parseInt(splitted[1], 10) || "";
+                this.model.target.tag = splitted[2] || "";
+                this.model.target.target = splitted[3] || "";
```

Comment:
Well you're never sure that you're not making an out of bound error then, maybe you could check if `splitted[1]`, `splitted[2]` and `splitted[3]` exist then : `splitted.length >= 4`